### DISCUSSION TITLE
Is there a way to use AND logic when combining multiple retention rules?

### DISCUSSION THREAD

**Discussion by hyeyoung-leee** [goharbor/harbor#20386] [Q&A] [answered]
Hi,

In the following case, I want to combine multiple retention rules with AND logic.

For all artifacts in the project a,
Rule 1: retain the artifacts pulled within the last 15 days
Rue 2: retain the most recently pulled 15 artifacts

It was confirmed that although the number of artifacts was less than 15, the images were cleaned up because more than 15 days had passed since the last pull date.

How can I resolve?

**Comment by MinerYang** [comment, marked as answer]
Hi @hyeyoung-leee 
For current behavior ,when you apply multiple rules to a repository, they are applied with OR logic rather than with AND logic, in this way  there is no prioritization of application of the rules on a given repository.

**Comment by hyeyoung-leee** [comment]
Hi, @MinerYang 
Thank you for your answer.
Do you have plans to use 'AND logic' or provide priority of policies?

**Reply by MinerYang** [reply]
We would like to transfer this discussion to a requirement issue.
